static/js/modules/top-entities.js

```diff
@@ -0,0 +1,204 @@
+  this.$elm = $(elm);
+  this.type = type;
+  this.category = this.$elm.data('category');
+  this.cycle = this.$elm.data('cycle');
+
+  this.$table = this.$elm.find('.js-top-table');
+  this.$dates = this.$elm.find('.js-dates');
+  this.$previous = this.$elm.find('.js-previous');
+  this.$next = this.$elm.find('.js-next');
+  this.$pageInfo = this.$elm.find('.js-page-info');
+  this.init();
+
+  $('.js-cycle').on('change', this.handleCycleChange.bind(this));
+  this.$elm.find('.js-category').on('change', this.handleCategoryChange.bind(this));
+  this.$elm.find('.js-previous').on('click', this.handlePagination.bind(this, 'previous'));
+  this.$elm.find('.js-next').on('click', this.handlePagination.bind(this, 'next'));
+}
+
+TopEntities.prototype.init = function() {
+  if (candidateCategories.indexOf(this.category) > -1) {
+    this.basePath = ['candidates', 'totals'];
+  } else {
+    this.basePath = ['totals', this.category];
+  }
+  this.baseQuery = {
+    sort: '-' + this.type,
+    per_page: 10,
+    sort_hide_null: true,
+    cycle: this.cycle
+  };
+  this.maxValue = Number(this.$table.find('.value-bar').first().data('value'));
+
+  // Store the current query for use in pagination and more
+  this.currentQuery = this.baseQuery;
+
+  // If it's a candidate table, add the office to the current query
+  if (candidateCategories.indexOf(this.category) > -1) {
+    this.office = this.category;
+    this.currentQuery.office = this.office;
+    this.category = 'candidates';
+  }
+
+  if (!this.currentQuery.page) {
+    this.$previous.addClass('is-disabled');
+  }
+
+  this.drawBars();
+};
+
+TopEntities.prototype.handleCycleChange = function(e) {
+  e.preventDefault();
+  this.cycle = e.target.value;
+  if (this.category === 'candidates') {
+      this.currentQuery = _.extend({}, this.baseQuery, {
+      cycle: this.cycle,
+      office: this.office,
+      page: 1
+    });
+  } else {
+    this.currentQuery = _.extend({}, this.baseQuery, {
+      cycle: this.cycle,
+      page: 1
+    });
+  }
+  this.loadData(this.currentQuery);
+  this.updateDates();
+};
+
+TopEntities.prototype.handleCategoryChange = function(e) {
+  e.preventDefault();
+  var category = e.target.value;
+  if (candidateCategories.indexOf(category) > -1) {
+    this.basePath = ['candidates', 'totals'];
+    this.category = 'candidates';
+    this.office = category;
+    this.currentQuery = _.extend({}, this.baseQuery, {
+      office: this.office,
+      cycle: this.cycle,
+      page: 1
+    });
+   } else {
+    this.basePath = ['totals', category];
+    this.category = category;
+    this.currentQuery = _.extend({}, this.baseQuery, {
+      cycle: this.cycle,
+      page: 1
+    });
+  }
+  this.loadData(this.currentQuery);
+};
+
+TopEntities.prototype.handlePagination = function(direction) {
+  var currentPage = this.currentQuery.page || 1;
+  if (direction === 'next') {
+    this.currentQuery.page = currentPage + 1;
+    this.$previous.removeClass('is-disabled');
```

Comment:
Ah! Good point. No reason other than there's often ~50 or more pages so unlikely to reach the end.